### DISCUSSION TITLE
Attached interfaces?

### DISCUSSION THREAD

**Discussion by Qiu233** [dotnet/csharplang#7216] [Language Ideas] [answered]
Will it be possible to have attached interfaces for `C#`, just like `trait` of Rust and `class` of Haskell?
This could be very useful for some features, especially the `auto implementation of interface`, which is similimar to Haskell's `DeriveGeneric`.

**Comment by 333fred** [comment, marked as answer]
That will be covered by a future phase of https://github.com/dotnet/csharplang/issues/5497. See "Phase C" (which is currently just a placeholder, but the idea is very much traits/type classes).
